File: WorldFeed/Server/src/WorldFeed.Common/Streaming/IStreamTrackManager.cs
namespace WorldFeed.Common.Streaming
{
    using System;
    using System.Collections.Generic;

    public interface IStreamTrackManager<T> : ITrackManager<T>, ITrackStringAnalyzer
    {
        /// <summary>
        /// Collection of tracks and their related actions. An action is invoked if 
        /// a track has been matched.
        /// </summary>
        List<Tuple<string, Action<T>>> GetMatchingTracksAndActions(string inputs);
    }
}

File: WorldFeed/Client/src/app/core/Core/Streaming/IStreamTrackManager.ts
import {Action} from "../../../c#-objects/TypeScript.NET-Core/packages/Core/source/FunctionTypes";
import {ITrackManager} from "./ITrackManager";
import {ITrackStringAnalyzer} from "./IStringAnalyzer";

export interface IStreamTrackManager<T> extends ITrackManager<T>, ITrackStringAnalyzer {
  /// Collection of tracks and their related actions. An action is invoked if a track has been matched.
  GetMatchingTracksAndActions(inputs: string): Array<Tuple<string, (t: T) => void>>;
}
